src/main/java/org/gwtproject/tools/cldr/CurrencyListProcessor.java

```diff
@@ -0,0 +1,366 @@
+        }
+        return ClassName.bestGuess("CurrencyList" + Processor.localeSuffix(parent, "_"));
+    }
+
+    private String getLocaleDefaultCurrency(GwtLocale locale) {
+        if ("default".equals(locale.getAsString())) {
+            return "USD";
+        }
+        if (isNull(locale.getRegion())) {
+            return regionsCurrencies.get(languageMostPopulatedRegion.get(locale.getLanguage()));
+        }
+        return regionsCurrencies.get(locale.getRegion());
+    }
+
+    private void generatePropertiesFile(String path, GwtLocale locale, Map<String, String> map, String[] keys) throws IOException {
+        System.out.println("Generating currency data for locale : " + locale);
+        PrintWriter pw = createOutputFile(path + "CurrencyData"+Processor.localeSuffix(locale) + ".properties");
+        printHeader(pw);
+        printVersion(pw, locale, "# ");
+
+        for (String key : keys) {
+            pw.print(key);
+            pw.print(" = ");
+            pw.println(map.get(key));
+        }
+        pw.close();
+    }
+
+    private void loadLocaleIndependentCurrencyData() {
+        InputFile supp = cldrFactory.getSupplementalData();
+
+        // load the table of default # of decimal places and rounding for each currency
+        defaultCurrencyFraction = 0;
+        XPathParts parts = new XPathParts();
+        for (String path : supp.listPaths("//supplementalData/currencyData/fractions/info")) {
+            parts.set(supp.getFullXPath(path));
+            Map<String, String> attr = parts.findAttributes("info");
+            if (attr == null) {
+                continue;
+            }
+            String curCode = attr.get("iso4217");
+            int digits = Integer.valueOf(attr.get("digits"));
+            if ("DEFAULT".equalsIgnoreCase(curCode)) {
+                defaultCurrencyFraction = digits;
+            } else {
+                currencyFractions.put(curCode, digits);
+            }
+            int roundingDigits = Integer.valueOf(attr.get("rounding"));
+            if (roundingDigits != 0) {
+                rounding.put(curCode, roundingDigits);
+            }
+        }
+
+        // find which currencies are still in use in some region, everything else
+        // should be marked as deprecated
+        for (String path : supp.listPaths("//supplementalData/currencyData/region")) {
+            parts.set(supp.getFullXPath(path));
+            Map<String, String> attr = parts.findAttributes("currency");
+            if (attr == null) {
+                continue;
+            }
+            String region = parts.findAttributeValue("region", "iso3166");
+            String curCode = attr.get("iso4217");
+            if ("ZZ".equals(region) || "false".equals(attr.get("tender")) || "XXX".equals(curCode)) {
+                // ZZ is an undefined region, XXX is an unknown currency code (and needs
+                // to be special-cased because it is listed as used in Anartica!)
```

Comment:
Antarctica

Reply:
this was copy paste, i wanted the readers to see the same comment as i am doing the same thing, here, i will fix it.